src/annotations/components/annotationForm/AnnotationForm.test.tsx

```diff
@@ -0,0 +1,226 @@
+import React from 'react'
+import {renderWithRedux} from 'src/mockState'
+import {
+  AnnotationForm,
+  END_TIME_IN_FUTURE_MESSAGE,
+  START_TIME_IN_FUTURE_MESSAGE,
+  TIMES_ARE_SAME_MESSAGE,
+  WRONG_ORDER_MESSAGE,
+} from 'src/annotations/components/annotationForm/AnnotationForm'
+import {fireEvent} from '@testing-library/react'
+import {createDateTimeFormatter} from 'src/utils/datetime/formatters'
+import {REQUIRED_ERROR} from 'src/annotations/components/annotationForm/AnnotationTimeInput'
+
+const setup = (annoType: 'point' | 'range', startTime, endTime) => {
+  const doNothing = () => {}
+
+  return renderWithRedux(
+    <AnnotationForm
+      title="Add"
+      type={annoType}
+      startTime={startTime}
+      endTime={endTime}
+      eventPrefix="testing"
+      onClose={doNothing}
+      onSubmit={doNothing}
+    />
+  )
+}
+
+const setupPoint = time => {
+  return setup('point', time, null)
+}
+
+const setupRange = (start, end) => {
+  return setup('range', start, end)
+}
+
+describe('Annotation Form Component tests', () => {
+  const fiveMinutesAgo = new Date().getTime() - 1000 * 60 * 5
+  const oneHourAgo = new Date().getTime() - 1000 * 60 * 60
+  const oneHourInFuture = fiveMinutesAgo + 60 * 60 * 1000
```

Comment:
dealing with actual time in tests is a kind of dangerous thing, especially around new years and daylight savings switch-overs.

i think it might be safer to just pick an arbitrary date in the past and use that as `now` and make all other dates relative to that. it could be as complicated as mocking out `Date.now` and calling that (which is identical to `new Date().getTime()`) or as simple as saying

```ts
const now = 1626135535000
const fiveMinutesAgo = now - 1000 * 60 * 5
```

etc

Reply:
the issue is that one of the things I am testing is that future dates are illegal/invalid.  new years/ dst/est switchovers are rare enough it isn't a problem for this one test.

Reply:
when future times aren't being tested this will be fine,
but when testing the future; it won't work.

Reply:
>when future times aren't being tested this will be fine,
>but when testing the future; it won't work.

how does the code determine if the date is in the future? does it calculate the future based on the time now, e.g. `Date.now()` / `new Time().getTime()` ? or does it calculate the future based on the time it's comparing e.g. `isInTheFuture = endDate > startDate`?

if it's the former, we'll have to mock out what time we return from `Date.now()` / `new Date()`

>new years/ dst/est switchovers are rare enough it isn't a problem for this one test.

I'm going to push back here and say it's not acceptable to have a situation where a test will knowingly break 2-3 times a year. Not sure if that's what we're looking at, but if that's the case, we can do better.

Reply:
for future testing, it does a compare to see if the time to check is After moment()
which gives it the current time.

i will come up with a scheme for injecting/mocking this after lunch.